### PR TITLE
More pythonic implementation of wolf sheep

### PR DESCRIPTION
**Pull request by quaquel**
This is a more Pythonic and single-file implementation of Wolf Sheep for benchmarking.

The single file is easier for profiling and line profiling.

### CODE REVIEW

File: benchmarks/WolfSheep/wolf_sheep.py [hunk 1]
```diff
@@ -8,15 +8,13 @@
     Center for Connected Learning and Computer-Based Modeling,
     Northwestern University, Evanston, IL.
 """
-import math
 
-from mesa import Model, Agent
+from mesa import Agent, Model
```
Comment (rht) [inline benchmarks/WolfSheep/wolf_sheep.py:12]: These 3 lines can be condensed to be just `import mesa`, then calling them would be `mesa.Agent`, `mesa.Model`, `mesa.space.MultiGrid`, in line with `np.array` or `pd.DataFrame`. See https://github.com/projectmesa/mesa-examples/blob/479eaf389e1a98bda0bdba5aa8f24ede0d84e764/examples/wolf_sheep/wolf_sheep/model.py#L18.

Comment (quaquel) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: but that is slower

**Comment by EwoutH** [comment]
Thanks! Would it be possible to split into two commits: One that makes it a single file, and then the actual changes? This diff is very difficult to review.

**Comment by quaquel** [comment]
There is no real point looking at the diff here. It's just a clean reimplementation.

File: benchmarks/WolfSheep/wolf_sheep.py (hunk 1 above)
Comment (rht) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: It's about the same:
```
In [1]: import mesa

In [2]: m = mesa.Model()

In [3]: %timeit mesa.Agent(0, m)
627 ns ± 125 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [4]: from mesa import Agent, Model

In [5]: m2 = Model()

In [6]: %timeit Agent(0, m2)
627 ns ± 148 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

Comment (rht) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: Except for the import lines, this PR LGTM.

Comment (quaquel) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: I really don't have strong opinions on this so if others want to change this as well, fine. 

Note, however, that the Schelling model already uses the same style of importing as I have used here. I also personally prefer _e.g._, `SchellingAgent(Agent)` over `SchellingAgent(mesa.Agent)`. This is really just a matter of taste.

Comment (rht) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: The convention in the [examples repo](https://github.com/projectmesa/mesa-examples/blob/479eaf389e1a98bda0bdba5aa8f24ede0d84e764/examples/wolf_sheep/wolf_sheep/model.py#L18) all uses `mesa.x`. The ones in the Agents.jl benchmark code are from the outdated version of the examples.

> This is really just a matter of taste.

But in the case of Agents.jl benchmark, they count and compare the LOC.

Comment (quaquel) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: > The convention in the [examples repo](https://github.com/projectmesa/mesa-examples/blob/479eaf389e1a98bda0bdba5aa8f24ede0d84e764/examples/wolf_sheep/wolf_sheep/model.py#L18) all uses mesa.x.

I ment our [benchmarks](https://github.com/projectmesa/mesa/blob/2837cc15fd0017523ea58358950d3eb32d1ad421/benchmarks/Schelling/schelling.py)

> But in the case of Agents.jl benchmark, they count and compare the LOC.

And we agree that this is only a rough proxy we should not obsess about.

Comment (rht) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: > I ment our [benchmarks](https://github.com/projectmesa/mesa/blob/2837cc15fd0017523ea58358950d3eb32d1ad421/benchmarks/Schelling/schelling.py)

The 3 versions of the code (mesa-examples, core Mesa benchmark, Agents.jl) ideally need to be kept in sync.

Comment (quaquel) [reply benchmarks/WolfSheep/wolf_sheep.py:12]: I agree. Once this is merged, I'll happily make PRs on the other libraries.

**Comment by tpike3** [comment]
As the build is failing at SugarScape_g1mt, I am going to merge.

**Pull request merged**